internal/statestore/redis_test.go

```diff
@@ -247,6 +357,515 @@ func TestConnect(t *testing.T) {
+}
+
+func TestCreateTicket(t *testing.T) {
+	cfg, closer := createRedis(t, true, "")
+	defer closer()
+	service := New(cfg)
+	require.NotNil(t, service)
+	defer service.Close()
+	ctx := utilTesting.NewContext(t)
+
+	var testCases = []struct {
+		description     string
+		ticket          *pb.Ticket
+		expectedCode    codes.Code
+		expectedMessage string
+	}{
+		{
+			description: "ok",
+			ticket: &pb.Ticket{
+				Id:         "1",
+				Assignment: &pb.Assignment{Connection: "2"},
+			},
+			expectedCode:    codes.OK,
+			expectedMessage: "",
+		},
+		{
+			description:     "nil ticket passed, err expected",
+			ticket:          nil,
+			expectedCode:    codes.Internal,
+			expectedMessage: "failed to marshal the ticket proto, id: : proto: Marshal called with nil",
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+		t.Run(tc.description, func(t *testing.T) {
+			err := service.CreateTicket(ctx, tc.ticket)
+			if tc.expectedCode == codes.OK {
+				require.NoError(t, err)
+			} else {
+				require.Error(t, err)
+				require.Equal(t, tc.expectedCode.String(), status.Convert(err).Code().String())
+				require.Contains(t, status.Convert(err).Message(), tc.expectedMessage)
+			}
+		})
+	}
+
+	// pass an expired context, err expected
+	ctx, cancel := context.WithCancel(context.Background())
+	cancel()
+	service = New(cfg)
+	err := service.CreateTicket(ctx, &pb.Ticket{
+		Id:         "222",
+		Assignment: &pb.Assignment{Connection: "2"},
+	})
+	require.Error(t, err)
+	require.Equal(t, codes.Unavailable.String(), status.Convert(err).Code().String())
+	require.Contains(t, status.Convert(err).Message(), "CreateTicket, id: 222, failed to connect to redis:")
+}
+
+func TestGetTicket(t *testing.T) {
+	cfg, closer := createRedis(t, false, "")
+	defer closer()
+	service := New(cfg)
+	require.NotNil(t, service)
+	defer service.Close()
+
+	ctx := utilTesting.NewContext(t)
+
+	err := service.CreateTicket(ctx, &pb.Ticket{
+		Id:         "mockTicketID",
+		Assignment: &pb.Assignment{Connection: "2"},
+	})
+	require.NoError(t, err)
+
+	c, err := redis.Dial("tcp", fmt.Sprintf("%s:%s", cfg.GetString("redis.hostname"), cfg.GetString("redis.port")))
+	require.NoError(t, err)
+	_, err = c.Do("SET", "wrong-type-key", "wrong-type-value")
+	require.NoError(t, err)
+
+	var testCases = []struct {
+		description     string
+		ticketID        string
+		expectedCode    codes.Code
+		expectedMessage string
+	}{
+		{
+			description:     "ticket is found",
+			ticketID:        "mockTicketID",
+			expectedCode:    codes.OK,
+			expectedMessage: "",
+		},
+		{
+			description:     "empty id passed, err expected",
+			ticketID:        "",
+			expectedCode:    codes.NotFound,
+			expectedMessage: "Ticket id: not found",
+		},
+		{
+			description:     "wrong id passed, err expected",
+			ticketID:        "123456",
+			expectedCode:    codes.NotFound,
+			expectedMessage: "Ticket id:123456 not found",
```

Comment:
Do we need a whitespace, but that could do this in a separate PR.
```suggestion
			expectedMessage: "Ticket id: 123456 not found",
```

Reply:
ok